File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        int x = 0;
        int y = 0;
        
        for (int[] s : sizes) {
            int a = s[0];
            int b = s[1];
            
            a = Math.max(s[0], s[1]);
            b = Math.min(s[0], s[1]);
            
            x = Math.max(x,a);
            y = Math.max(y,b);
        }
        
        answer = x * y;
        
        return answer;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int a = 0;
    int b = 0;
    
    for (vector<int> size : sizes) {
        sort(size.begin(), size.end());
        a = max(a, size[0]);
        b = max(b, size[1]);
    }
    
    answer = a * b;
    
    
    
    return answer;
}